pyterrier/batchretrieve.py

```diff
@@ -62,11 +63,70 @@ def __init__(self, verbose=0, **kwargs):
         super().__init__(kwargs)
         self.verbose = verbose
 
+def _from_dataset(dataset : Union[str,Dataset], 
+            clz,
+            variant : str = None, 
+            version='latest',            
+            **kwargs):
+
+    from . import get_dataset
+    from .io import autoopen
+    import os
+    import json
+    
+    if isinstance(dataset, str):
+        dataset = get_dataset(dataset)
+    if version != "latest":
+        raise ValueError("index versioning not yet supported")
+    indexref = dataset.get_index(variant)
+
+    classname = clz.__name__
+    # now look for, e.g., BatchRetrieve.args.json file, which will define the args for BatchRetrieve, e.g. stemming
```

Comment:
Will naming this BatchRetrieve.args.json make it annoying if/when we change the name of the class later?

Reply:
I think we can just make an additional json file and run that for a while.